scene/2d/physical_bone_2d.cpp

```diff
@@ -0,0 +1,320 @@
+			} else {
+				current_parent = nullptr;
+			}
+		}
+	}
+}
+
+void PhysicalBone2D::_find_joint_child() {
+	for (int i = 0; i < get_child_count(); i++) {
+		Node *child_node = get_child(i);
+		Joint2D *potential_joint = Object::cast_to<Joint2D>(child_node);
+		if (potential_joint) {
+			child_joint = potential_joint;
+			break;
+		}
+	}
+}
+
+String PhysicalBone2D::get_configuration_warning() const {
+	String warning = Node2D::get_configuration_warning();
+
+	if (!parent_skeleton) {
+		if (warning != String()) {
+			warning += "\n\n";
+		}
+		warning += TTR("A PhysicalBone2D only works with a Skeleton2D or another PhysicalBone2D as a parent node!");
+	}
+
+	if (parent_skeleton && bone2d_index <= -1) {
+		if (warning != String()) {
+			warning += "\n\n";
+		}
+		warning += TTR("A PhysicalBone2D needs to be assigned to a Bone2D node in order to function! Please set a Bone2D node in the inspector.");
+	}
+
+	if (!child_joint) {
+		PhysicalBone2D *parent_bone = Object::cast_to<PhysicalBone2D>(get_parent());
+		if (parent_bone) {
+			if (warning != String()) {
+				warning += "\n\n";
+			}
+			warning += TTR("A PhysicalBone2D node should have a Joint2D-based child node to keep bones connected! Please add a Joint2D-based node as a child to this node!");
+		}
+	}
+
+	return warning;
+}
+
+void PhysicalBone2D::_auto_configure_joint() {
+	if (!auto_configure_joint) {
+		return;
+	}
+
+	if (child_joint) {
+		// Node A = parent | Node B = this node
+		Node *parent_node = get_parent();
+		PhysicalBone2D *potential_parent_bone = Object::cast_to<PhysicalBone2D>(parent_node);
+
+		if (potential_parent_bone) {
+			child_joint->set_node_a(child_joint->get_path_to(potential_parent_bone));
+			child_joint->set_node_b(child_joint->get_path_to(this));
+		} else {
+			WARN_PRINT("Cannot setup joint without a parent PhysicalBone2D node.");
+		}
+
+		// Place the child joint at this node's position.
+		child_joint->set_global_position(get_global_position());
+	}
+}
+
+void PhysicalBone2D::_start_physics_simulation() {
+	if (_internal_simulate_physics) {
+		return;
+	}
+
+	// Reset to Bone2D position
+	_position_at_bone2d();
+
+	// Let the RigidBody executing its force integration.
+	if (!follow_bone_when_simulating) {
+		PhysicsServer2D::get_singleton()->body_set_force_integration_callback(get_rid(), this, "_direct_state_changed");
+	}
+
+	// Apply the layers and masks
+	PhysicsServer2D::get_singleton()->body_set_collision_layer(get_rid(), get_collision_layer());
+	PhysicsServer2D::get_singleton()->body_set_collision_mask(get_rid(), get_collision_mask());
+
+	_internal_simulate_physics = true;
+}
+
+void PhysicalBone2D::_stop_physics_simulation() {
+	if (_internal_simulate_physics) {
+		// Stop the RigidBody from executing its force integration.
+		PhysicsServer2D::get_singleton()->body_set_force_integration_callback(get_rid(), nullptr, "");
+		_internal_simulate_physics = false;
+
+		// Reset to Bone2D position
+		_position_at_bone2d();
+	}
+}
```

Comment:
Please follow this to properly turn the physics on and off: https://github.com/godotengine/godot/blob/master/scene/3d/physics_body_3d.cpp#L2574

Reply:
I'm confused, the code you linked to is what I'm doing. Can you elaborate?